x-pack/elastic-agent/docs/elastic-agent-dynamic-inputs.asciidoc

```diff
@@ -0,0 +1,837 @@
+[[dynamic-input-configuration]]
+[role="xpack"]
+= Dynamic input configuration
+
+//TODO: Break into multiple source files to make this info easier to change over
+//time. Maybe move to a separate directory.
+
+//PH: I'm not sure what the best heading for this topic is. What are users
+//likely to search on?
```

Comment:
I had to double-check my initial [draft](https://gist.github.com/ph/0e28b537ecc24820bfaa34b184173203), This is hard, our idea was to reduce the wording used and just goes with **Input Configuration** and hide the "dynamic" nature of the input.

 Do you think this is OK?

Reply:
@mostlyjason Wdyt about that?

Reply:
The page title could be input configuration, and it'd be nice to have a header for conditions. They might search on it because its a key in the YAML file. It also might be a term that beats autodiscover users would search on.

Reply:
Nevermind I see you have a section on conditions below

Reply:
The problem with "input configuration" is that it raises the expectation that the section will cover everything about configuring inputs, but it doesn't. It only really talks about variables and setting the input configuration based on conditions.

When we roll out the new doc system, we'll be able to add keywords, but for now, I think I should mention autodiscovery somewhere in the text so that existing users can find this topic.

Reply:
I think mentioning autodiscovery ++

Reply:
++ good to mention in the text